### PR TITLE
Homework#1 done

### PR DESCRIPTION
**Pull request by ivan-sheremeta**
Homework#1 done, V=8
I. Sheremeta

### CODE REVIEW

File: homework.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
-VARIANT = 0
+VARIANT = 8
 
 def generate
-  # put your solution here (delete the line below)
-  rand(1000000).to_s + '-' + rand(1000000).to_s
+  letters = *("A".."Z")
+
+  letter1, letter2 = 2.times.map { letters.sample(3).join }
```
Comment (sveredyuk) [inline homework.rb]: var1, var2.... not ruby-way :)

File: homework.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
+  num1, num2 = 2.times.map { rand(100..999).to_s }
```
Comment (sveredyuk) [inline homework.rb]: same

File: homework.rb (hunk 1 above)
Comment (ivan-sheremeta) [reply homework.rb]: do you mean parallel assignment of variables?

Comment (sveredyuk) [reply homework.rb]: I mean the name of variables. Think how to rewrite whole code to avoid creating vars with nums

Comment (ivan-sheremeta) [reply homework.rb]: no variables just long oneliner is ok? 
just tested the speed is better.
`[letters.sample(3).join, rand(100..999).to_s, letters.sample(3).join, rand(100..999).to_s].join("-")`

Comment (sveredyuk) [reply homework.rb]: in fact you don't need vars.

Try this and explain to me how it works:
```ruby
def generate
  letters = *("A".."Z")

  letters = -> { letters.sample(3).join }
  numbers = -> { rand(100..999).to_s }

  [letters.call, numbers.call, letters.call, numbers.call].join("-")
end
```

Comment (ivan-sheremeta) [reply homework.rb]: THNQ for the detailed prompt.
As I understand, this works due to stabby lambda (lambda literal) syntax. 
In result array, lambda are just called with no args. 
TBH, I`ve rarely used that.

Comment (sveredyuk) [reply homework.rb]: Yes, And Lambda gives you runtime calls. Lambda/Proc/Block are very cool and widely used construction. Please, pay attention to learn them good enough.

**Pull request closed without merging**